csrc/quantization/awq/gemm_kernels.cu

```diff
@@ -334,7 +334,7 @@ __global__ void __launch_bounds__(64)
   }
 
   // TODO: Shang: Hoist loop invariance.
-  for (int ax1_0_1 = 0; ax1_0_1 < 4; ++ax1_0_1) {
+  for (int ax1_0_1 = 0; ax1_0_1 < (N / 32); ++ax1_0_1) {
```

Comment:
Should we add an assert in awq.py that the N dim is divisible by 32?

Reply:
N is the template parameter, not the output dimension of weight.